syntax = "proto2";

package apollo.data.fuel;

import "apps/k8s/spark_submitter/spark_submit_arg.proto";

// Will be translated to SparkSubmitArg with pre-defined workflow.
message SaasJobArg {
  enum JobType {
    UNKNOWN = 0;
    CONTROL_PROFILING = 1;
    PERCEPTION_MODEL_TRAINING = 2;
    SENSOR_CALIBRATION = 3;
    VIRTUAL_LANE_GENERATION = 4;
    VEHICLE_CALIBRATION = 5;
    OPEN_SPACE_PLANNER_PROFILING = 6;
    PREDICTION_MODEL_TRAINING = 7;
    DYNAMIC_MODELING = 8;
  }
  optional JobType job_type = 1;

  optional Partner partner = 2;
  map<string, string> flags = 3;
}

